mesh/ncmesh.cpp

```diff
@@ -950,8 +1120,14 @@ void NCMesh::CheckIsoFace(int vn1, int vn2, int vn3, int vn4,
    }
 }
 
+void NCMesh::SetNodeScale(int p0, int p1, real_t scale)
+{
+   Node* node = nodes.Find(p0, p1);
+   if (node) { node->SetScale(GetScale(scale, p0 > p1)); }
```

Comment:
If setting scale on a node, and its not found, that would probably be an assertable error? Or is speculatively setting the scale helpful?

Reply:
It is definitely possible to create a disaster, although I expect this function to be called only after a Node is found (that is the usage so far). I added an MFEM_ASSERT in https://github.com/mfem/mfem/commit/e5cfc35f87969a015c88f892f8927267bd9ef4c5 so that performance won't be affected in an optimized build.

Reply:
The assert fails in some cases, so I removed it in 95a5c503b10a44d919ad498575311606924c5539. It fails when called for certain pairs of midpoints, e.g. `ncmesh.cpp:1456`: `SetNodeScale(mid01, mid23, scale_y);`. It is not always known what pairs of midpoints have a `Node` defined, but when the `Node` does exist, its scale should be set.